src/Zipper.elm

```diff
@@ -0,0 +1,81 @@
+module Zipper exposing
+    ( Zipper
+    , getCurrent
+    , makeZipper
+    , moveBackwards
+    , moveBackwardsTo
+    , moveBackwardsWhile
+    , moveForward
+    , moveForwardTo
+    , moveForwardWhile
+    )
+
+import Debug
+import List exposing (append, head, isEmpty, tail)
+import List.Extra as ListE
+import Maybe exposing (withDefault)
+
+
+type alias Zipper a =
+    { previous : List a
+    , current : a
+    , next : List a
+    }
+
+
+makeZipper : List a -> a -> List a -> Maybe (Zipper a)
+makeZipper previous current next =
+    Just { previous = previous, current = current, next = next }
```

Comment:
Why return `Maybe (Zipper a)` if you are guaranteed to successfully create one?

Reply:
What if the input is `Maybe List    Nothing    Maybe List`?

Reply:
But this could happen only if you receive a `Maybe List`, `Maybe a` and `Maybe List`

Reply:
Hmm..  I don't know then 🤔
We entered it's type together on Friday,
guess I'll delete- and we'll add it later if needed.

Reply:
Now that I come to think about it, `a` is used as a generic type. So it might be a `Maybe Int` and in that case you could get `[]`, `Nothing` and `[]`. That is the worst case I came up with. But you still always return `Just Zipper`.

Reply:
Maybe I should make it that if the `current` is Nothing it'll return `Nothing`?

Reply:
We had a typo on Friday. It should return a `Zipper a`, not a `Maybe (Zipper a)`. You can see it simply by realizing you don't ever fail (never return a `Nothing`).

Reply:
Just saw that it was not right of me to delete also the `Maybe`'s of the other functions (other than `makeZipper`).

Reply:
Indeed. Only the constructor and the destructors (i.e. `getCurrent`, `getNext`, `getPrevious`) cannot  fail. All others can fail, and thus must return a `Maybe x`.

Reply:
Another way of looking at it is that `Maybe x` has a very nice property of being able to represent possible failure.
This naturally leads to functions as `toRightEnd : Zipper a -> Zipper a` which traverses the `Zipper a` to the right end.
How does it "know" it's reached the end? It's inner `moveRight : Zipper a -> Maybe (Zipper a)` fails with a `Nothing`.
If you have a `moveRight : Zipper a -> Zipper a`, you take away the ability to represent failure, and make functions such as `toRightEnd` much less nice to implement.

Challenge: What other types you're already familiar with have this nice property?